Assets/Scripts/SimpleCoin.cs

```diff
@@ -5,13 +5,46 @@ namespace DefaultNamespace
 {
     public class SimpleCoin : MonoBehaviour
     {
+        [SerializeField] private int _value = 1;
         [SerializeField] private string _playerTag = "Player";
+        [SerializeField] private float _movementTime = 1f;
+        [SerializeField] private float _movementSpeed = 1f;
+        [SerializeField] private float _rotationsSpeed = 1f;
         private void OnTriggerEnter(Collider other)
         {
             if (other.CompareTag(_playerTag))
             {
+                _coinCounter.Add(_value);
                 Destroy(gameObject);
             }
         }
+
+        private void Start()
+        {
+            _rigidbody = GetComponent<Rigidbody>();
+            _coinCounter = GameObject.FindWithTag("GameController").GetComponent<CoinCounter>();
+        }
+
+        private void FixedUpdate()
+        {
+            Vector3 angularVelocity = new Vector3(0, _rotationsSpeed, 0);
```

Comment:
a viable solution, but a good approach is to NOT use physics when possible (because of potential physics bugs). An easier way would be to just animate the object's Y position via Mathf.Sin